### DISCUSSION TITLE
Any specific reason to include 'table' class to tsr model, even if there is a separate detection model?

### DISCUSSION THREAD

**Discussion by sajan-gohil** [microsoft/table-transformer#37] [Q&A] [answered]
There is a model for table detection but the table class has been added to the model for tsr as well? Is this just experimental or do they serve different purpose?

**Comment by bsmock** [comment, marked as answer]
Thanks for raising this great point of discussion. If we train both a TD model and a TSR model only on the PubTables-1M dataset, then the "table" class is not needed in the TSR model. 

But we include it in the TSR model for the additional flexibility it brings in potential future uses of the model.

For example, while ideally every image fed to the TSR model contains only one table, there's no reason why this must be the case. Including the "table" class means that the current TSR model could be adapted to cases where more than one table appears in an image fed to it.
